### ISSUE TITLE
Uncaught TypeError: Cannot read property 'data' of null

### ISSUE THREAD

**Issue by AdamSkaidon** [Ziink/atom-nav-panel#2]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.2.3
**System**: Mac OS X 10.11.1
**Thrown From**: [nav-panel](https://github.com/Ziink/atom-nav-panel) package, v0.0.6

### Stack Trace

Uncaught TypeError: Cannot read property 'data' of null

```
At /Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:132

TypeError: Cannot read property 'data' of null
    at NavView.module.exports.NavView.getPanelState (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:132:17)
    at NavView.module.exports.NavView.saveFileState (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:200:14)
    at /Users/Adam/.atom/packages/nav-panel/lib/nav-panel.coffee:84:18
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at PaneContainer.module.exports.PaneContainer.willDestroyPaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane-container.js:285:27)
    at Pane.module.exports.Pane.destroyItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:442:17)
    at Pane.module.exports.Pane.destroyActiveItem (/Applications/Atom.app/Contents/Resources/app.asar/src/pane.js:429:12)
    at Workspace.module.exports.Workspace.destroyActivePaneItem (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:639:35)
    at Workspace.module.exports.Workspace.destroyActivePaneItemOrEmptyPane (/Applications/Atom.app/Contents/Resources/app.asar/src/workspace.js:678:21)
    at atom-workspace.commandRegistry.add.core:close (/Applications/Atom.app/Contents/Resources/app.asar/src/register-default-commands.js:170:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:533:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:356:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:99:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
     -2:47.8.0 core:cancel (atom-text-editor.editor.mini.is-focused)
     -2:40.7.0 tree-view:add-file (span.name.icon.icon-file-directory)
  3x -2:38.5.0 core:backspace (atom-text-editor.editor.mini.is-focused)
  2x -2:33.5.0 editor:select-to-beginning-of-word (atom-text-editor.editor.mini.is-focused)
  3x -2:29 editor:move-to-beginning-of-word (atom-text-editor.editor.mini.is-focused)
     -2:26 core:confirm (atom-text-editor.editor.mini.is-focused)
     -2:10.6.0 core:paste (atom-text-editor.editor.is-focused)
     -2:07.2.0 atom-beautify:beautify-editor (atom-text-editor.editor.is-focused)
     -2:01.9.0 core:copy (atom-text-editor.editor.is-focused)
     -1:58.7.0 core:paste (atom-text-editor.editor.is-focused)
     -1:47.4.0 core:save (atom-text-editor.editor.is-focused)
     -1:10.6.0 tree-view:add-folder (li.file.entry.list-item.status-modified.selected)
     -1:08.5.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:58.7.0 tree-view:cut (li.file.entry.list-item.status-added.selected)
     -0:53.5.0 tree-view:paste (span.name.icon.icon-file-directory)
     -0:02.2.0 core:close (ol.tree-view.list-tree.has-collapsable-children.focusable-panel.full-menu)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
MagicPython, v0.5.7
antialiased, v0.1.0
atom-beautify, v0.28.19
atom-material-syntax, v0.2.5
atom-material-ui, v0.8.0
autoclose-html, v0.19.0
autocomplete-paths, v1.0.2
file-icons, v1.6.12
linter, v1.11.2
linter-csslint, v1.1.0
minimap, v4.16.1
nav-panel, v0.0.6
open-in-browsers, v0.0.7
pigments, v0.19.0
tool-bar-main, v0.0.8
wakatime, v5.0.1

# Dev
No dev packages
```

**Comment by Ziink** [comment]
Fixed in v0.0.7

**Comment by AdamSkaidon** [comment]
Happened again:

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.2.4
**System**: Mac OS X 10.11.1
**Thrown From**: [nav-panel](https://github.com/Ziink/atom-nav-panel) package, v0.0.8


### Stack Trace

Uncaught TypeError: Cannot read property 'data' of null

```
At /Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:430

TypeError: Cannot read property 'data' of null
    at NavView.module.exports.NavView.destroyPanel (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:430:38)
    at NavView.module.exports.NavView.updateFile (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:125:6)
    at /Users/Adam/.atom/packages/nav-panel/lib/nav-panel.coffee:75:22
```

### Commands

```
     -8:45 core:backspace (atom-text-editor.editor.is-focused)
     -8:35 core:copy (atom-text-editor.editor.is-focused)
  2x -8:33.4.0 core:paste (atom-text-editor.editor.is-focused)
     -8:29.2.0 core:save (atom-text-editor.editor.is-focused)
     -8:27.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -8:26.3.0 core:save (atom-text-editor.editor.is-focused)
  2x -8:13.3.0 core:backspace (atom-text-editor.editor.is-focused)
     -8:11.9.0 core:save (atom-text-editor.editor.is-focused)
  2x -1:37.7.0 core:backspace (atom-text-editor.editor.is-focused)
     -1:32.3.0 core:save (atom-text-editor.editor.is-focused)
     -0:31.8.0 core:backspace (atom-text-editor.editor.is-focused)
     -0:31.4.0 core:save (atom-text-editor.editor.is-focused)
  2x -0:04.6.0 core:undo (atom-text-editor.editor.is-focused)
     -0:02.0 core:redo (atom-text-editor.editor.is-focused)
     -0:01.6.0 core:save (atom-text-editor.editor.is-focused)
     -0:01.6.0 core:close (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
MagicPython, v0.5.7
antialiased, v0.1.0
atom-beautify, v0.28.19
atom-material-syntax, v0.2.6
atom-material-ui, v0.8.0
autoclose-html, v0.19.0
autocomplete-paths, v1.0.2
file-icons, v1.6.12
linter, v1.11.3
linter-csslint, v1.1.0
minimap, v4.16.2
nav-panel, v0.0.8
open-in-browsers, v0.0.7
pigments, v0.19.1
tool-bar-main, v0.0.8
wakatime, v5.0.1

# Dev
No dev packages
```

**Comment by AdamSkaidon** [comment]
Happened in latest version of Aom, just closing files this time.

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.3.1
**System**: Mac OS X 10.11.2
**Thrown From**: [nav-panel](https://github.com/Ziink/atom-nav-panel) package, v0.0.10


### Stack Trace

Uncaught TypeError: Cannot read property 'data' of null

```
At /Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:430

TypeError: Cannot read property 'data' of null
    at NavView.module.exports.NavView.destroyPanel (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:430:38)
    at NavView.module.exports.NavView.updateFile (/Users/Adam/.atom/packages/nav-panel/lib/nav-view.coffee:125:6)
    at /Users/Adam/.atom/packages/nav-panel/lib/nav-panel.coffee:75:22
```

### Commands

```
     -9:57.6.0 autocomplete-plus:confirm (atom-text-editor.editor.is-focused.autocomplete-active)
     -9:50.5.0 core:backspace (atom-text-editor.editor.is-focused)
     -9:49.6.0 core:move-down (atom-text-editor.editor.is-focused)
  3x -9:49.2.0 editor:select-to-beginning-of-word (atom-text-editor.editor.is-focused)
     -9:47.6.0 core:paste (atom-text-editor.editor.is-focused)
     -9:42.9.0 core:copy (atom-text-editor.editor.is-focused)
     -9:41.2.0 editor:newline (atom-text-editor.editor.is-focused)
  5x -9:40.1.0 core:backspace (atom-text-editor.editor.is-focused)
     -9:32.4.0 editor:newline (atom-text-editor.editor.is-focused)
     -9:32.2.0 core:paste (atom-text-editor.editor.is-focused)
     -9:26.2.0 core:save (atom-text-editor.editor.is-focused)
  6x -0:12.3.0 core:close (atom-text-editor.editor.is-focused)
```

### Config

```json
{
  "core": {
    "audioBeep": false,
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ]
  }
}
```

### Installed Packages

```coffee
# User
MagicPython, v0.5.7
antialiased, v0.1.0
atom-beautify, v0.28.19
atom-material-syntax, v0.2.10
atom-material-ui, v0.8.0
autoclose-html, v0.19.0
autocomplete-paths, v1.0.2
caniuse, v0.5.5
file-icons, v1.6.13
imdone-atom, v1.3.11
js-hyperclick, v1.4.1
linter, v1.11.3
linter-csslint, v1.1.0
minimap, v4.18.0
nav-panel, v0.0.10
neon-cursor, v1.4.0
open-in-browsers, v0.0.7
pigments, v0.19.3
tool-bar-main, v0.0.8
wakatime, v5.0.2

# Dev
No dev packages
```

**Comment by Ziink** [comment]
Was happening after saving newly created file. Fixed in v0.0.11
